Refactor verify(), extract verifyCredential().

dmitrizagidulin opened the pull request:
Fixes issue #53.

Also, adds a section on `challenge` param to the README, as requested in https://github.com/digitalbazaar/vc-js/issues/39.

dlongley commented on CHANGELOG.md:
```diff
@@ -1,5 +1,15 @@
+- `verifiableCredential` param is now optional in `createPresentation()`.
+- **BREAKING**: `verify()` now only verifies presentations, not credentials,
+  (since that will be the most common use case). For credentials, create a
+  separate `verifyCredential()` method.
```
```suggestion
  (since that will be the most common use case). For credentials, a separate
  `verifyCredential()` method has been added.
```

dlongley commented on README.md:
```diff
@@ -394,6 +380,41 @@ const result = await vc.verify({
+### Verifying a Verifiable Credential
+
+For most situations, you will only need to verify a Verifiable Presentation,
+since Verifiable Credentials that are not wrapped in a VP are a rare occurrence.
```
```suggestion
For most situations, Verifiable Credentials will be wrapped in a Verifiable
Presentation and the entire VP should be verified. However, this library
provides a utility function to verify a Verifiable Credential on its own.
```

dlongley commented on README.md:
```diff
@@ -394,6 +380,41 @@ const result = await vc.verify({
+
+* If you're using a custom `@context`, make sure it's resolvable
+* You're using the correct public key and corresponding suite
+* Your Controller Document is reachable via a `documentLoader`
```
Hmm, I think this whole section needs a bit of a rewrite. I started to rework it through suggestions here:

```suggestion
* If you're using a custom `@context`, make sure it's resolvable,
  providing a custom `documentLoader` as needed
* You're passing the cryptographic suite(s) your application supports
  (be prepared to handle errors if the VC does not use a matching suite)
* You have specified a specific verification method (e.g., public key) that
  your application will accept via a suite *or* you provide a custom
  `documentLoader` that can resolve acceptable verification methods
* Controller Document is reachable via a `documentLoader`
```

But it got unwieldy. I think you need to say that you should provide a `documentLoader` that can:

* Resolve any acceptable JSON-LD contexts
* Resolve any acceptable verification methods
* Resolve any acceptable controller documents for those verification methods

It's strange to talk about providing a public key to the suite or to "pick the right suite" for the VC. That's not generally how verification software should be written. Instead, you should be specifying the suites the application supports and a document loader that will only load the things that are acceptable to the application. Then you pass the VC and see if it meets your criteria or not.

dmitrizagidulin replied:
I agree - I figured we can come back and expand the docs, in a separate pr.

dmitrizagidulin commented:
@dlongley I've incorporated the suggestions in your comments. Also, simplified the Background section (moved most of it to a separate document), per @gannan08's earlier suggestion.
